lib/generators/solidus_paypal_commerce_platform/install/install_generator.rb

```diff
@@ -31,26 +30,42 @@ def install_solidus_backend_support
         end
       end
 
-      def install_solidus_frontend_support
-        support_code_for('solidus_frontend') do
+      def install_solidus_starter_frontend_support
+        solidus_frontend_is_missing = !Bundler.locked_gems.specs.map(&:name).include?('solidus_frontend')
+
+        # We can't do better than this for now, over time we might want to mark things differently
+        support_code_for('solidus_starter_frontend', run_if: solidus_frontend_is_missing) do
```

Comment:
What about taking the frontend as an option for the installer? The main Solidus installer could delegate it. The `run_if` heuristic could fail, as I might want to install the extension with no frontend.